pkg/etcd-backup-operator/controller/controller.go

```diff
@@ -0,0 +1,72 @@
+package controller
+
+import (
+	"context"
+	"time"
+
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+func (b *Backup) run(ctx context.Context) {
+	source := cache.NewListWatchFromClient(
+		b.backupCRCli.EtcdV1beta2().RESTClient(),
+		api.EtcdBackupResourcePlural,
+		b.namespace,
+		fields.Everything(),
+	)
+
+	b.queue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "etcd-backup-operator")
+	b.indexer, b.informer = cache.NewIndexerInformer(source, &api.EtcdBackup{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc:    b.onAdd,
+		UpdateFunc: b.onUpdate,
+		DeleteFunc: b.onDelete,
+	}, cache.Indexers{})
+
+	defer b.queue.ShutDown()
+
+	logrus.Info("starting backup controller")
+	go b.informer.Run(ctx.Done())
+
+	if !cache.WaitForCacheSync(ctx.Done(), b.informer.HasSynced) {
+		logrus.Error("Timed out waiting for caches to sync")
```

Comment:
This is not a timeout error, right?

Reply:
The doc says 
```
// WaitForCacheSync waits for caches to populate.  It returns true if it was successful, false
// if the controller should shutdown
```

Maybe `logrus.Error("cache failed to sync.")` is more appropriate? cc/ @hasbro17

Reply:
Why even error? "It should shutdown" from the docs.

Reply:
should we log the reason of shutting done?

Reply:
Yes. We should.
But it's not relevant here. Right?